lib/pkcs7.js

```diff
@@ -341,13 +341,14 @@ p7.createSignedData = function() {
      * Includes the signature and certificate without the signed data.
      */
     signDetached: function() {
-      msg.baseSign(true);
+      msg._baseSign(true);
     },
 
     /**
      * Signs the content.
+     * @param detached If signing should be done in detached mode
      */
-    baseSign: function(detached) {
+    _baseSign: function(detached) {
```

Comment:
@vbuch I don't know what the future holds for `_baseSign`.  I really don't like the look of `msg._baseSign(true)`.  How do you feel about the signature being `msg._baseSign({detached: true})`, and for the default case, it can just be `msg._baseSign()`?

Reply:
Would introduce a couple of lines of more code, but would look much better, yes. Will do it.

Reply:
Hm... I'm thinking now that I see the code... baseSign could be needless in this case. As long as sing() has an optional `options` parameter we could call sign() and sign({detached: true}) instead of sign() and signDetached() What do you think @mattcollier ?

Reply:
@vbuch good point.  It will be up to @dlongley and @davidlehn to answer this question.